google/cloud/documentai_toolbox/wrappers/document.py

```diff
@@ -573,13 +573,21 @@ def from_batch_process_metadata(
         if metadata.state != documentai.BatchProcessMetadata.State.SUCCEEDED:
             raise ValueError(f"Batch Process Failed: {metadata.state_message}")
 
-        return [
-            Document.from_gcs(
-                *gcs_utilities.split_gcs_uri(process.output_gcs_destination),
-                gcs_input_uri=process.input_gcs_source,
+        documents: List[Document] = []
+        for process in metadata.individual_process_statuses:
+            gcs_bucket, gcs_prefix = gcs_utilities.split_gcs_uri(
+                process.output_gcs_destination
             )
-            for process in list(metadata.individual_process_statuses)
-        ]
+            documents.append(
+                Document.from_gcs(
+                    gcs_bucket_name=gcs_bucket,
+                    # Trailing slash added because it's missing from output_gcs_destination.
+                    # https://github.com/googleapis/python-documentai-toolbox/issues/271
+                    gcs_prefix=f"{gcs_prefix}/",
```

Comment:
This feels like it will be difficult to maintain...

Does this imply that callers of `Document.from_gcs` should always include a trailing slash in `gcs_prefix`?  This is not consistent with the docstring here: https://github.com/googleapis/python-documentai-toolbox/blob/main/google/cloud/documentai_toolbox/wrappers/document.py#L495-L498

Can we let the caller pass in whatever, and in the implementation of `Document.from_gcs` we strip all trailing slashes, and add it only when calling GCS?